datalad/distributed/drop.py

```diff
@@ -0,0 +1,804 @@
+    execute a dataset update before dropping dataset components.
+    """
+    _examples_ = [
+        {'text': "Drop single file content",
+         'code_py': "drop('path/to/file')",
+         'code_cmd': "datalad drop <path/to/file>"},
+        {'text': "Drop all file content in the current dataset",
+         'code_py': "drop('.')",
+         'code_cmd': "datalad drop"},
+        {'text': "Drop all file content in a dataset and all its subdatasets",
+         'code_py': "drop(dataset='.', recursive=True)",
+         'code_cmd': "datalad drop -d <path/to/dataset> -r"},
+        {'text': "Disable check to ensure the configured minimum number of "
+                 "remote sources for dropped data",
+         'code_py': "drop(path='path/to/content', reckless='availability')",
+         'code_cmd': "datalad drop <path/to/content> --reckless availability"},
+        {'text': 'Drop (uninstall) an entire dataset '
+                 '(will fail with subdatasets present)',
+         'code_py': "drop(what='all')",
+         'code_cmd': "datalad drop --what all"},
+        {'text': 'Kill a dataset recklessly with any existing subdatasets too'
+                 '(this will be fast, but will disable any and all safety '
+                 'checks)',
+         'code_py': "drop(what='all', reckless='kill', recursive=True)",
+         'code_cmd': "datalad drop --what all, --reckless kill --recursive"},
+    ]
+
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            metavar="DATASET",
+            doc="""specify the dataset to perform drop from.
+            If no dataset is given, the current working directory is used
+            as operation context""",
+            constraints=EnsureDataset() | EnsureNone()),
+        path=Parameter(
+            args=("path",),
+            metavar="PATH",
+            doc="path of a dataset or dataset component to be dropped",
+            nargs="*",
+            constraints=EnsureStr() | EnsureNone()),
+        reckless=Parameter(
+            args=("--reckless",),
+            doc="""disable individual or all data safety measures that would
+            normally prevent potentially irreversible data-loss.
+            With 'modification', unsaved modifications in a dataset will not be
+            detected. This improves performance at the cost of permitting
+            potential loss of unsaved or untracked dataset components.
+            With 'availability', detection of dataset/branch-states that are
+            only available in the local dataset, and detection of an
+            insufficient number of file-content copies will be disabled.
+            Especially the latter is a potentially expensive check which might
+            involve numerous network transactions.
+            With 'undead', detection whether a to-be-removed local annex is
+            still known to exists in the network of dataset-clones is disabled.
```

Comment:
A few small typos in this sentence:

"With 'undead', detection of whether a to-be-removed local annex is still known to exist in the network of dataset-clones is disabled."

add "of" and change "exists" -> "exist"

Reply:
Fixed locally. Will push when nothing else is reported. Thx!